lab/network/macvlan.py

```diff
@@ -84,9 +84,11 @@ def addresses(self):
             if addr.is_link_local:
                 continue
             elif addr.version == 4:
-                data.setdefault(socket.AF_INET, []).append(addr)
+                values = data.setdefault(socket.AF_INET, [])
+                values.append(addr), values.append(prefix)
```

Comment:
You can't do this. This is a list for a reason - its possible to have multiple addresses per network interface. You can't just shove whatever extra data you want in here just because.

Reply:
`addr` is already an `ipaddress.IPv4Address`/`ipaddress.IPv6Address` - we use `ipaddress.ip_address` to parse it.

It would make much more sense to just mixin the prefix information and use `ipaddress.ip_interface` instead.

Reply:
Actually, we probably should just mark this method as deprecated and introduce a new method that returns interfaces instead of addresses.